### PR TITLE
Echo with env variable

### PR DESCRIPTION
**Pull request by momeyer**
# buffer.c
- init_buffer will just initialize the buffer struct, remove garbage and set index to 0
- append_char_to_buffer appends the current char to buffer and increment arg start and buffer index
- append_value_to_buffer gets the value of env variable and append it to buffer. Also moves the arg start until the end of `$VAR_NAME` and increment buffer index
# echo_command.c
- handle spaces will check cur char, if it's a space it append one space to buffer if index != 0 and skip all the spaces.
# parse_quotes.c
- Added functions related to quotes here

### CODE REVIEW

File: src/commands/buffer.c [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+t_arg	append_char_to_buffer(t_arg arg, t_buffer *buffer)
+{
+	buffer->buf[buffer->index] = *arg.start;
+	++(buffer->index);
+	++(arg.start);
+	return (arg);
+}
+
+void	append_value_to_buffer(t_arg *echo_arg, t_buffer *buffer)
```
Comment (Ryno95) [inline src/commands/buffer.c:20]: ...env_value....();

File: src/env/env_utils.c [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+	int	len;
+
+	len = 0;
+	while (key[len] != '\0' && !isspace(key[len])
+		&& !is_quote(key[len]) && !is_variable(key[len]))
+		++len;
+	return (len);
+}
+
+t_bool	is_variable(char cur)
```
Comment (Ryno95) [inline src/env/env_utils.c:18]: is_env_var();

Comment (Ryno95) [reply src/env/env_utils.c:18]: also check for space at the end

File: src/env/env_utils.c [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+{
+	return (cur == VARIABLE);
```
Comment (Ryno95) [inline src/env/env_utils.c:20]: VARIABLE_TOKEN

File: src/env/environment.c [hunk 4]
```diff
@@ -59,17 +60,18 @@ void	display_env(t_env *env, t_output_stdout output)
 	}
 }
 
-t_env	*find_variable(t_env *env, const char *key_name)
+t_env	*find_variable(t_env *env, const char *key)
 {
-	const int	name_len = ft_strlen(key_name) + 1;
-	int			i;
+	const size_t	name_len = get_key_len(key);
```
Comment (Ryno95) [inline src/env/environment.c:65]: name_len could be key_len

File: tests/acceptance/echo_feature_bash.sh [hunk 5]
```diff
@@ -47,7 +47,11 @@ printTestName "ECHO"
 runWithoutQuotes "echo Hello" "echo Hello" "Simple String"
 runWithoutQuotes "echo Hello      Hello" "echo Hello      Hello" "String with spaces to be trimmed"
 runWithoutQuotes "echo -n-n Hello" "echo -n-n Hello" "String with invalid n flag"
+runWithoutQuotes "echo $PWD" "echo \$PWD" "With env var"
+runWithQuotes 'Hello      Hello $PWD' "echo 'Hello      Hello \$PWD'" "String with single quotes"
 runWithQuotes "Hello      Hello" "echo \"Hello      Hello\"" "Quoted string, shouldn't trim"
+runWithQuotes "Hello      Hello $PWD " "echo \"Hello      Hello \$PWD \"" "Quoted string, shouldn't trim and should get var name"
+runWithQuotes "Hello      Hello $PWDs " "echo \"Hello      Hello \$PWDs \"" "Quoted string, shouldn't trim, and variable is invalid"
 runWithQuotes "                Hello      Hello" "echo \"                Hello      Hello\"" "Quoted string, shouldn't trim"
```
Comment (Ryno95) [inline tests/acceptance/echo_feature_bash.sh:56]: The single $ var token ;)) is being skipped when printing 
to be added to test:
`echo $`
`echo $USER$`

**Review by Ryno95 — CHANGES_REQUESTED** [review]
looks awesome, glad we ran through it in person, comments for some changes have been added as well as an edge case

**Review by Ryno95 — APPROVED** [review]
Did my best to break it, no use ttying to break something that is full-proof ;)
